Scripts/Runtime/Entities/Lifecycle/EntityDestroySystem.cs

```diff
@@ -0,0 +1,199 @@
+using Anvil.Unity.DOTS.Data;
+using Anvil.Unity.DOTS.Jobs;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Entities;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// System that helps in destroying <see cref="Entity"/>'s
+    /// </summary>
+    /// /// <remarks>
+    /// By default, this system updates in <see cref="SimulationSystemGroup"/> but can be configured by subclassing
+    /// and using the <see cref="UpdateInGroupAttribute"/> to target a different group.
+    /// 
+    /// By default, this system updates before <see cref="EntitySpawnSystem"/> so that Entity reuse in a chunk can
+    /// occur. This can be configured by subclassing and using the <see cref="UpdateBeforeAttribute"/> or
+    /// <see cref="UpdateAfterAttribute"/>.
+    /// 
+    /// By default, this system uses the <see cref="EndSimulationEntityCommandBufferSystem"/> to playback the
+    /// generated <see cref="EntityCommandBuffer"/>s. This can be configured by subclassing and using the
+    /// <see cref="UseCommandBufferSystemAttribute"/> to target a different <see cref="EntityCommandBufferSystem"/>
+    /// </remarks>
+    [UpdateInGroup(typeof(SimulationSystemGroup))]
+    [UpdateBefore(typeof(EntitySpawnSystem))]
+    [UseCommandBufferSystem(typeof(EndSimulationEntityCommandBufferSystem))]
+    public class EntityDestroySystem : AbstractAnvilSystemBase
```

Comment:
> Having read through this system I'm not sure the purpose of it. Why not just have people write their destroy operations to ECBs?

It's the yang to `EntitySpawnSystem`s yin. I agree it's not super heavy lifting like the spawning is, but for discoverability, you see the Spawn System and then think "ok how do I destroy..." and there it is, intuitively named and accessed the same way. 

You could do custom ECBs but then you are getting the specific ECB system, creating a new ECB, doing your work and then remembering to add as producer to the ECB system. 

This wraps all that cruft for you. 

> If an application is trying to coordinate all entity destruction of the certain type to a specific phase then that's on the application to encourage? I don't know if destroying ALL entities in the same phase is a common practice.

The application can do that. We just provide a happy default that minimizes the amount of main thread sync time. 
If the application wants to support different phases, they simply subclass and use the subclassed system. 

> If I'm understanding correctly this is for scenarios where there a large number of entities being created and destroyed in one frame and the optimization prevents unity from releasing a chunk by cramming all create and delete operations into one command buffer.

No, this is more of an API convenience but it does scale and minimizes main thread time. 
You could be spawning / destroying a handful or vast amounts in a frame. 
There is NO sharing of buffers between Destroy and Spawning. (And in Spawning, each type is it's own Buffer. No more ECB.ParallelWriter, it's not necessary). 
The DestroySystem does go first to remove entities so that when Spawning happens, we can fill up the same chunk and avoid a second allocation. 

> 
>     Unity will immediately release a chunk back into the pool at the end of the command buffer's playback if all entities are destroyed in a chunk?
>     Chunk allocation is a somewhat expensive operation? My understanding is that it shouldn't be since the source of chunk memory is a pre-allocated pool.
>     Chunks won't be released mid way through an ECB's playback.

Good questions, someone with some R&D time can check that out in the future. The benefits of this aren't related but might be a happy coincidence.

Reply:
> It's the yang to EntitySpawnSystems yin. I agree it's not super heavy lifting like the spawning is, but for discoverability, you see the Spawn System and then think "ok how do I destroy..." and there it is, intuitively named and accessed the same way.

The purpose of anvil isn't to create a new abstraction for people that don't want to learn ECS. The purpose is to extend the functionality. If we're not providing benefit beyond discover-ability then I'm not sure, at an anvil level, why we'd want this.

> You could do custom ECBs but then you are getting the specific ECB system, creating a new ECB, doing your work and then remembering to add as producer to the ECB system.
>
> This wraps all that cruft for you.

This doesn't seem a whole lot different than grabbing/using/releasing an access handle. Same number of calls?

Reply:
Decided to move destroy into application and see if this is still useful.

Reply:
Done